qt/Torrent.h

```diff
@@ -320,6 +320,21 @@ class Torrent : public QObject
         return s ? static_cast<double>(s - l) / static_cast<double>(s) : 0.0;
     }
 
+    [[nodiscard]] constexpr double seedRatioPercentDone() const noexcept
+    {
+        auto const r = ratio();
+        auto const g = getSeedRatioLimit();
+        return static_cast<double>(r) / static_cast<double>(g.value_or(r));
```

Comment:
- I think both of these casts can be removed?  `r` is already a `double` and `g` is already a `std::optional<double>`
- This method calls a non-`constexpr` func `getSeedRatioLimit()` so this can't be `constexpr` either
- Need to safeguard against divide-by-zero here

Reply:
1. Oops yeah, not sure why i kept that.
2. Oh right, thank you.
3. :scream: